be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -81,24 +81,21 @@ class BinaryPlainPageBuilder : public PageBuilder {
     Slice finish() override {
         _finished = true;
 
-        size_t offsets_pos = _buffer.size();
-
-        // Set up the header
-        encode_fixed32_le(&_buffer[0], _offsets.size());
-        encode_fixed32_le(&_buffer[4], offsets_pos);
-
-        _buffer.append(&_offsets[0], _offsets.size() * sizeof(uint32_t));
+        // Set up trailer
+        for (int i = 0; i < _offsets.size(); i++) {
+            put_fixed32_le(&_buffer, _offsets[i]);
```

Comment:
Why not use variable length encode?

Reply:
It is convenient to decode offset array when using fixed length to store offset value,otherwise setting a new array to record start position of offset array?

Reply:
Because offset bytes(4) is usually small compared to data bytes, so use varint to store offsets won't save too much space. And use fixed-size offsets enable us to avoid allocating offsets array and decoding varints during read. AFAIK, store offsets in plain old little endian ints are common in data stores such as LevelDB, Parquet, ORC, Elasticsearch, etc.

Reply:
Because offset bytes(4) is usually small compared to data bytes, so use varint to store offsets won't save too much space. And use fixed-size offsets enable us to avoid allocating offsets array and decoding varints during read. AFAIK, store offsets in plain old little endian ints are common in data stores such as LevelDB, Parquet, ORC, Elasticsearch, etc.

Reply:
OK, I See